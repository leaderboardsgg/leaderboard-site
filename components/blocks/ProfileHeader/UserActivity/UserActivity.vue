<template>
  <div class="user-activity" v-bind="$attrs">
    <p><span>Runs: </span>{{ runCount }}</p>
    <p><span>Games: </span>{{ gameCount }}</p>
  </div>
</template>

<script lang="ts">
import { defineComponent } from '@nuxtjs/composition-api';

export default defineComponent({
  name: 'UserActivity',
  props: {
    gameCount: {
      default: () => 6,
      type: Number,
    },
    runCount: {
      default: () => 42,
      type: Number,
    },
  },
});
</script>

<style scoped>
.user-activity {
  & span {
    @apply font-semibold;
  }
}
</style>
