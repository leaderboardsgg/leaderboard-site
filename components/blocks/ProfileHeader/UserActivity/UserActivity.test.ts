import { mountSuspended } from '@nuxt/test-utils/runtime'
import UserActivity from './UserActivity.vue'

describe('<UserActivity />', () => {
  it('should render without crashing', async () => {
    const wrapper = await mountSuspended(UserActivity, {
      props: {
        gameCount: 69,
        runCount: 420,
      },
    })

    expect(wrapper.isVisible()).toBe(true)
  })
})
