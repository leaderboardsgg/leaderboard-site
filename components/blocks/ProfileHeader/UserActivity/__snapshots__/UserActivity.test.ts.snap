// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserActivity /> renders correctly 1`] = `
<div
  class="user-activity"
>
  <p>
    <span>
      Runs: 
    </span>
    42
  </p>
   
  <p>
    <span>
      Games: 
    </span>
    6
  </p>
</div>
`;
